src/compiler/ir/README.md

```diff
@@ -0,0 +1,162 @@
+region's entrypoint. The compiler therefore needs to do some work to find the
+entrypoint given the specified parent.
+
+## `CallStmt`
+
+Call a method, on the specified receiver. The method is identified by its name.
+At compile-time, the method's selector should be obtained from the given name.
+
+In order to dispatch the call, in the general case, the selector should be used
+to index into the object's vtable, to retrieve the address of the target.
+Optimizations may enable us to dispatch the call statically, using for example,
+the type of the receiver.
+
+The method being called could be static or not (ie. does it accept a `self`
+parameter, not to be confused with the dispatch strategy). In either case, the
+receiver of the call is always passed as the first argument to the method.  In
+the case of a static method call, the receiver may not be an actual object, but
+only a pointer to a descriptor, as created by a `StaticTypeStmt`.
+
+## `WhenStmt`
+
+Schedule a behaviour on a set of cown. The `closure_index` corresponds to the
+index of the body within the `MethodIR`.
+
+This corresponds to a `rt::Cown::schedule` call. The message sent to the cowns
+should provide storage for a pointer to the closure's body, the list of cowns
+and the captures.
+
+The closure's body will accept as many parameters as there are cowns and
+captures specified. When invoking the body, each cown argument should be
+transformed into a reference to its contents, before being passed as an
+argument. The captures should be passed as is.
+
+## `ReadFieldStmt`
+
+Read a field from an object. The field is identified by its name. The general
+case is similar to method calls: the field's selector is used to look up the
+field's offset in the object's vtable.
+
+## `WriteFieldStmt`
+
+Write a field into an object. The procedure for locating the offset of the
+field is identical to a field read.
+
+When writing a reference counted value to an object in a region that is
+equipped with a RememberedSet, the value should be inserted into the RS.
+Otherwise, the reference count should be incremented.
+
+## `IntegerLiteralStmt`, `StringLiteralStmt`, `UnitStmt`
+
+These create values from a literal. The support for strings is very special
+cased for demonstration purposes.
+
+## `MatchBindStmt`
+
+This is an unchecked cast, used at the top of the target of a match statement.
+
+## `ViewStmt`
+
+If the argument to the statement is an owned value, the statement produces an
+unowned reference to the same object. Otherwise this is equivalent to a copy.
+
+## `EndScopeStmt`
+
+This statement is inserted whenever variables go out of scope. This may not be
+the end of a BasicBlock, if a block expression is used for example.
+
+This is where a value's "release" code should be executed. This could be
+`rt::Region::release`, `rt::Cown::release` or `rt::Immutable::release`, for
+owned, cown and immutable values, respectively. Unowned values do not need to
+be released.
```

Comment:
This is just related to lexical blocks, not liveness analysis, right?

Reply:
Correct. Liveness isn't encoded in the IR, although it can be accessed in `analysis->liveness` if you need it.